test/player_test.cpp

```diff
@@ -29,13 +29,13 @@ int RunBlockTest(int frames, int flags)
 }
 
 #define NORM 0
-#define BAL ISPL_FASTRECOVER
-#define STA ISPL_FASTERRECOVER
-#define HAR ISPL_FASTESTRECOVER
-#define BALSTA (ISPL_FASTRECOVER | ISPL_FASTERRECOVER)
-#define BALHAR (ISPL_FASTRECOVER | ISPL_FASTESTRECOVER)
-#define STAHAR (ISPL_FASTERRECOVER | ISPL_FASTESTRECOVER)
-#define ZEN (ISPL_FASTRECOVER | ISPL_FASTERRECOVER | ISPL_FASTESTRECOVER)
+#define BAL static_cast<uint32_t>(ItemSpecialEffect::FastHitRecovery)
+#define STA static_cast<uint32_t>(ItemSpecialEffect::FasterHitRecovery)
+#define HAR static_cast<uint32_t>(ItemSpecialEffect::FastestHitRecovery)
+#define BALSTA static_cast<uint32_t>(ItemSpecialEffect::FastHitRecovery | ItemSpecialEffect::FasterHitRecovery)
+#define BALHAR static_cast<uint32_t>(ItemSpecialEffect::FastHitRecovery | ItemSpecialEffect::FastestHitRecovery)
+#define STAHAR static_cast<uint32_t>(ItemSpecialEffect::FasterHitRecovery | ItemSpecialEffect::FastestHitRecovery)
+#define ZEN static_cast<uint32_t>(ItemSpecialEffect::FastHitRecovery | ItemSpecialEffect::FasterHitRecovery | ItemSpecialEffect::FastestHitRecovery)

```

Comment:
Could be worth changing these to constexpr values and making `BlockData` a collection of structs instead of a 2d int array. That'd save having to put the cast in these defines and then invert it inside `RunBlockTest`, could take `ItemSpecialEffect` as an arg there.

```cpp
struct BlockTestCase {
    int expectedRecoveryFrame;
    int maxRecoveryFrame;
    ItemSpecialEffect itemFlags;
}

BlockTestCase BlockData[] = { ... };
```

I wouldn't say it's necessary for this PR unless you wanted to have a go, definitely good for a followup in the near future.

Reply:
Done